src/core/core.scale.js

```diff
@@ -1170,27 +1166,27 @@ class Scale extends Element {
 	 * @private
 	 */
 	_drawGrid(chartArea) {
-		var me = this;
-		var gridLines = me.options.gridLines;
+		const me = this;
+		const gridLines = me.options.gridLines;
 
 		if (!gridLines.display) {
 			return;
 		}
 
-		var ctx = me.ctx;
-		var chart = me.chart;
-		var context = {
+		const ctx = me.ctx;
+		const chart = me.chart;
+		let context = {
 			scale: me,
 			tick: me.ticks[0],
 		};
-		var axisWidth = gridLines.drawBorder ? resolve([gridLines.borderWidth, gridLines.lineWidth, 0], context, 0) : 0;
-		var items = me._gridLineItems || (me._gridLineItems = me._computeGridLineItems(chartArea));
-		var width, color, i, ilen, item;
+		const axisWidth = gridLines.drawBorder ? resolve([gridLines.borderWidth, gridLines.lineWidth, 0], context, 0) : 0;
+		const items = me._gridLineItems || (me._gridLineItems = me._computeGridLineItems(chartArea));
+		let i, ilen;
 
 		for (i = 0, ilen = items.length; i < ilen; ++i) {
-			item = items[i];
-			width = item.width;
-			color = item.color;
+			const item = items[i];
```

Comment:
The V8 (or even babel) probably fixes this, but essentially we are instructing to allocate new variable for each iteration (that will be garbage collected later). So i think it would be good practice to allocate outside loop and reuse.
Correct me if I'm wrong :)

Reply:
I wouldn't think we'd be able to notice any impact here. Maybe if it were something we were doing for every data point, but there won't be many grid lines

Reply:
I agree that this is not a critical part. The amount of garbage generated is quite large and we should try to minimize that.

Reply:
Yeah, could be worth testing for some of the loops that go over all data points